### PR TITLE
Serialise instances for Proxy and Fingerprint

### PR DESCRIPTION
**Pull request by Shimuuar**
Here are instance for Proxy (in very unlikely case someone needs it)

I'm not sure Fingerprint is right one. It's some hash which is stored in two 64-bit words for convenience. Encoding it as bytestring makes more sense. What do you think?

### CODE REVIEW

**Comment by dcoutts** [comment]
@Shimuuar I think that encoding is just fine. Given that it uses the standard encoding scheme it can evolve in future.

File: tests/Tests/Serialise.hs [hunk 1]
```diff
@@ -175,6 +177,7 @@ testTree = testGroup "Serialise class"
       , mkTest (T :: T [Int])
       , mkTest (T :: T UTCTime)
       , mkTest (T :: T Version)
+      , mkTest (T :: T Fingerprint)
```
Comment (dcoutts) [inline tests/Tests/Serialise.hs]: This line seems to be failing to build in travis in one config. Could you have a look?

**Comment by dcoutts** [comment]
@Shimuuar so apart from the travis failure, this looks fine to me.

**Comment by Shimuuar** [comment]
Typeable instance for Fingerprint is missing in GHC 7.8. Well missing instances inevitably lead to weeping and gnashing of teeth. I've force-pushed updated patch

Also it looks like build with 8.0 times out.

**Pull request merged**